src/server/tiered_storage.cc

```diff
@@ -610,14 +689,26 @@ size_t TieredStorage::ReclaimMemory(size_t goal) {
 }
 
 bool TieredStorage::ShouldStash(const PrimeValue& pv) const {
+  // Check value state
+  if (pv.IsExternal() || pv.HasStashPending())
+    return false;
+
+  // Estimate value size
+  auto [size, rep] = DetermineSerializationParams(pv);
```

Comment:
please swap condition with line 702. No need to do size estimations if hash offloading is disabled.